integration/v7/isolated/labels_command_test.go

```diff
@@ -478,147 +478,105 @@ var _ = Describe("labels command", func() {
 				})
 			})
 		})
 
-		Describe("service-broker labels", func() {
-			var broker *servicebrokerstub.ServiceBrokerStub
+		Describe("service-instance labels", func() {
+			var serviceInstanceName string
 
 			BeforeEach(func() {
-				helpers.LoginCF()
-
 				spaceName = helpers.NewSpaceName()
 				helpers.SetupCF(orgName, spaceName)
-				broker = servicebrokerstub.Register()
+
+				serviceInstanceName = helpers.NewServiceInstanceName()
+				Eventually(helpers.CF("cups", serviceInstanceName)).Should(Exit(0))
 			})
 
 			AfterEach(func() {
 				helpers.QuickDeleteOrg(orgName)
-				broker.Forget()
 			})
 
-			When("there are labels set on the service broker", func() {
+			When("there are labels set on the service-instance", func() {
 				BeforeEach(func() {
-					session := helpers.CF("set-label", "service-broker", broker.Name, "some-other-key=some-other-value")
+					session := helpers.CF("set-label", "service-instance", serviceInstanceName, "some-other-key=some-other-value", "some-key=some-value")
 					Eventually(session).Should(Exit(0))
-
 				})
 
-				It("returns the labels associated with the broker", func() {
-					session := helpers.CF("labels", "service-broker", broker.Name)
-					Eventually(session).Should(Say(regexp.QuoteMeta("Getting labels for service-broker %s as %s...\n\n"), broker.Name, username))
+				It("lists the labels", func() {
+					session := helpers.CF("labels", "service-instance", serviceInstanceName)
+					Eventually(session).Should(Say(regexp.QuoteMeta("Getting labels for service-instance %s in org %s / space %s as %s...\n\n"), serviceInstanceName, orgName, spaceName, username))
 					Eventually(session).Should(Say(`key\s+value`))
-					Expect(session).To(Say(`some-other-key\s+some-other-value`))
+					Eventually(session).Should(Say(`some-key\s+some-value`))
+					Eventually(session).Should(Say(`some-other-key\s+some-other-value`))
 					Eventually(session).Should(Exit(0))
 				})
 			})
 
-			When("there are no labels set on the service broker", func() {
+			When("there are no labels set on the service-instance", func() {
 				It("indicates that there are no labels", func() {
-					session := helpers.CF("labels", "service-broker", broker.Name)
-					Eventually(session).Should(Say(regexp.QuoteMeta("Getting labels for service-broker %s as %s...\n\n"), broker.Name, username))
-					Expect(session).ToNot(Say(`key\s+value`))
-					Eventually(session).Should(Say("No labels found."))
+					session := helpers.CF("labels", "service-instance", serviceInstanceName)
 					Eventually(session).Should(Exit(0))
+					Expect(session).Should(Say(regexp.QuoteMeta("Getting labels for service-instance %s in org %s / space %s as %s...\n\n"), serviceInstanceName, orgName, spaceName, username))
+					Expect(session).ToNot(Say(`key\s+value`))
+					Expect(session).Should(Say("No labels found."))
 				})
 			})
 
-			When("the service broker does not exist", func() {
+			When("the app does not exist", func() {
```

Comment:
It should be "the service-instance" does not exist by looking at the whole file.